src/Factory/ActiveCampaign/ConnectionFactory.php

```diff
@@ -4,17 +4,24 @@
 
 namespace Webgriffe\SyliusActiveCampaignPlugin\Factory\ActiveCampaign;
 
-use Webgriffe\SyliusActiveCampaignPlugin\Model\ActiveCampaign\Connection;
 use Webgriffe\SyliusActiveCampaignPlugin\Model\ActiveCampaign\ConnectionInterface;
 
 final class ConnectionFactory implements ConnectionFactoryInterface
 {
+    public function __construct(
+        private string $connectionFQCN
```

Comment:
We could have used a generic "classFQCN" so it's easier to maintain among all Factories, you can even put it in an asbtract class or a trait and use this in all of them

@lruozzi9 @mmenozzi what do you think?

Reply:
Ok, but how can you handle all the arguments of the create method?

Reply:
Everything else remain the same. Take a look at my commit

Reply:
What I had in mind was to make a single factory class for all models. However, it becomes difficult to type the parameters and then do all the tests on these factories ... For the moment let's keep it that way!